packages/jaeger-ui/src/components/Monitor/ServicesView/index.tsx

```diff
@@ -102,142 +79,94 @@ export const spanKindOptions = [
+  fetchAllServiceMetrics: (serviceName: string, query: MetricsAPIQueryParams) => void;
+};
 
-  updateDimensions() {
-    if (this.graphDivWrapper.current) {
-      this.setState({
-        graphWidth: this.graphDivWrapper.current.offsetWidth - 24,
-      });
+export function MonitorATMServicesViewImpl({
+  services,
+  servicesLoading,
+  metrics,
+  fetchServices,
+  fetchAggregatedServiceMetrics,
+  fetchAllServiceMetrics,
+}: TProps) {
+  const [graphWidth, setGraphWidth] = useState(300);
+  const [serviceOpsMetrics, setServiceOpsMetrics] = useState<ServiceOpsMetrics[] | undefined>(undefined);
+  const [searchOps, setSearchOps] = useState('');
+  const [graphXDomain, setGraphXDomain] = useState<number[]>([]);
+  const [selectedService, setSelectedService] = useState(store.get('lastAtmSearchService') || '');
+  const [selectedSpanKind, setSelectedSpanKind] = useState<spanKinds>(
+    store.get('lastAtmSearchSpanKind') || 'server'
+  );
+  const [selectedTimeFrame, setSelectedTimeFrame] = useState(
+    store.get('lastAtmSearchTimeframe') || oneHourInMilliSeconds
+  );
+  const [endTime, setEndTime] = useState(Date.now());
+
+  const graphDivWrapper = useRef<HTMLDivElement>(null);
+  const docsLink = getConfigValue('monitor.docsLink');
+
+  const updateDimensions = () => {
+    if (graphDivWrapper.current) {
+      setGraphWidth(graphDivWrapper.current.offsetWidth - 24);
     }
-  }
-
-  handleServiceChange = (value: string) => {
-    this.setState({ selectedService: value }, () => {
-      trackSelectService(value);
-      this.fetchMetrics();
-    });
   };
 
-  handleSpanKindChange = (value: string) => {
-    this.setState({ selectedSpanKind: value as spanKinds }, () => {
-      const { label } = spanKindOptions.find(option => option.value === value)!;
-      trackSelectSpanKind(label);
-      this.fetchMetrics();
-    });
-  };
-
-  handleTimeFrameChange = (value: number) => {
-    this.setState({ selectedTimeFrame: value }, () => {
-      const { label } = timeFrameOptions.find(option => option.value === value)!;
-      trackSelectTimeframe(label);
-      this.fetchMetrics();
-      this.calcGraphXDomain();
-    });
+  const calcGraphXDomain = () => {
+    const currentTime = Date.now();
+    setGraphXDomain([currentTime - selectedTimeFrame, currentTime]);
   };
 
-  fetchMetrics() {
-    const { fetchAllServiceMetrics, fetchAggregatedServiceMetrics, services } = this.props;
-    const { selectedService, selectedSpanKind, selectedTimeFrame } = this.state;
+  const fetchMetrics = () => {
     const currentService = selectedService || services[0];
 
     if (currentService) {
-      this.endTime = Date.now();
+      const newEndTime = Date.now();
+      setEndTime(newEndTime);
+
       store.set('lastAtmSearchSpanKind', selectedSpanKind);
       store.set('lastAtmSearchTimeframe', selectedTimeFrame);
-      store.set('lastAtmSearchService', this.getSelectedService());
+      store.set('lastAtmSearchService', getSelectedService());
```

Comment:
This is setting the last selected service to `store` on selecting a new service. The expected behavior is to set the currently selected service to store